### PR TITLE
Refactor TS-486

### PR DESCRIPTION
**Pull request by codacy-vrhpires**

### CODE REVIEW

File: src/configCreator.ts [hunk 1]
```diff
@@ -1,114 +1,202 @@
...
+  const defaultFilesToAnalyze = [
+    srcDirPath + "/**/*.ts",
+    srcDirPath + "/**/*.tsx",
+    srcDirPath + "/**/*.js",
+    srcDirPath + "/**/*.jsx"
+  ]
+
+  //TODO: pass this result into optionsCreator mostly because of "options.useEslintrc"
+  const eslintrcExists = eslintrcExistsInSrcDir(srcDirPath)
+  const options = (codacyrc && eslintrcExists) ? {} : await optionsCreator(srcDirPath, tsconfigFile, codacyrc)
```
Comment (MiguelMarcelino) [inline src/configCreator.ts]: I am slightly concerned by this. Because if `eslintrcExistsInSrcDir` only checks whether there is a config file in the source directory, we might miss it in other directories. I get that it gets rid of the error, but is this the correct way to do it?

File: src/configCreator.ts [hunk 2]
```diff
@@ -1,114 +1,202 @@
...
+      }
+    }
+    else if (DEBUG) {
+      options.baseConfig.rules = patternsToRules(getAllPatterns())
+    }
+  }
+  options.cwd = srcDirPath
+  options.errorOnUnmatchedPattern = false
+  options.resolvePluginsRelativeTo = "/"
+  //TODO: take into consideration if there are config files in repo
```
Comment (MiguelMarcelino) [inline src/configCreator.ts]: If we search for configuration files in the repo here, will we not override the `patterns` field provided in `.codacyrc`?

File: src/docGeneratorMain.ts [hunk 3]
```diff
@@ -83,305 +76,280 @@ async function main() {
...
 
   console.log("Generate mocha description files")
   await docGenerator.downloadDocs(
-
     `${githubBaseUrl}/lo1tuma/eslint-plugin-mocha/master/docs/rules/`,
     "mocha"
   )
 
-  console.log("Generate no-unsanitized description files")
+  console.log("Generate n description files")
```
Comment (MiguelMarcelino) [inline src/docGeneratorMain.ts:217]: Is this change intended?

Comment (codacy-vrhpires) [reply src/docGeneratorMain.ts:217]: The "no-unsanitized" is kept.
Changed "node" to "n" because:
"forked from [eslint-plugin-node](https://www.npmjs.com/package/eslint-plugin-node) v11.1.0. as the original repository seems [no longer maintained](https://github.com/mysticatea/eslint-plugin-node/issues/300)."

File: src/configCreator.ts (hunk 1 above)
Comment (codacy-vrhpires) [reply src/configCreator.ts]: The TODO is actually a RFC for that exact concern. This also relates to line 173 (of this file).
We have to further discuss this point to get a clearer idea of what we want to happen for different scenarios.

File: src/configCreator.ts (hunk 2 above)
Comment (codacy-vrhpires) [reply src/configCreator.ts]: If we use them (config files across repo): yes. As it is (the flag is false): no.
This is the feedback I wanted and decisions we must make.

**Review by DMarinhoCodacy — APPROVED** [review]
